include/wil/registry_basic.h

```diff
@@ -1531,133 +1528,123 @@ inline HRESULT get_value_dword_nothrow(HKEY key, _In_opt_ PCWSTR value_name, _In
-    optional_value<::std::wstring> optionalvalue;
-    const auto hr = regview.try_get_value<::std::wstring>(subkey, value_name, optionalvalue);
-    RETURN_IF_FAILED(hr);
+    return get_value_byte_vector_nothrow(key, nullptr, value_name, type, data);
+}
 
-    if (optionalvalue.type != optional_value_status::has_value)
-    {
-        RETURN_WIN32(ERROR_FILE_NOT_FOUND);
-    }
+inline HRESULT get_value_string_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, ::std::wstring* data) WI_NOEXCEPT try
+{
+    const reg_view_details::reg_view_nothrow regview{key};
+    ::wil::reg::optional_value<::std::wstring> registry_value{ regview.try_get_value<::std::wstring>(subkey, value_name) };
+    RETURN_IF_FAILED(registry_value.inner_error);
 
-    std::swap(data, optionalvalue.value);
+    ::std::swap(*data, registry_value.value);
     return S_OK;
 }
 CATCH_RETURN()
 
-inline HRESULT get_value_string_nothrow(HKEY key, _In_opt_ PCWSTR value_name, ::std::wstring& data) WI_NOEXCEPT
+inline HRESULT get_value_string_nothrow(HKEY key, _In_opt_ PCWSTR value_name, ::std::wstring* data) WI_NOEXCEPT
 {
     return get_value_string_nothrow(key, nullptr, value_name, data);
 }
 
-inline HRESULT get_value_expanded_string_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, ::std::wstring& data) WI_NOEXCEPT try
+inline HRESULT get_value_expanded_string_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, ::std::wstring* data) WI_NOEXCEPT try
 {
-    const reg_view_details::reg_view_nothrow regview{key};
-    optional_value<::std::wstring> optionalvalue;
-    const auto hr = regview.try_get_value<::std::wstring>(subkey, value_name, optionalvalue, REG_EXPAND_SZ);
-    RETURN_IF_FAILED(hr);
+    const reg_view_details::reg_view_nothrow regview{ key };
+    ::wil::reg::optional_value<::std::wstring> registry_value{ regview.try_get_value<::std::wstring>(subkey, value_name, REG_EXPAND_SZ) };
+    RETURN_IF_FAILED(registry_value.inner_error);
 
-    if (optionalvalue.type != optional_value_status::has_value)
-    {
-        RETURN_WIN32(ERROR_FILE_NOT_FOUND);
-    }
-
-    std::swap(data, optionalvalue.value);
+    ::std::swap(*data, registry_value.value);
     return S_OK;
 }
 CATCH_RETURN()
 
-inline HRESULT get_value_expanded_string_nothrow(HKEY key, _In_opt_ PCWSTR value_name, ::std::wstring& data) WI_NOEXCEPT
+inline HRESULT get_value_expanded_string_nothrow(HKEY key, _In_opt_ PCWSTR value_name, ::std::wstring* data) WI_NOEXCEPT
 {
     return get_value_expanded_string_nothrow(key, nullptr, value_name, data);
 }
 
-inline HRESULT get_value_multistring_wstring_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, ::std::vector<::std::wstring>* return_value) WI_NOEXCEPT try
+inline HRESULT get_value_multistring_wstring_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, _In_ ::std::vector<::std::wstring>* return_value) WI_NOEXCEPT try

```

Comment:
These pointer params (including the ones that were formerly references) are all `_Out_`